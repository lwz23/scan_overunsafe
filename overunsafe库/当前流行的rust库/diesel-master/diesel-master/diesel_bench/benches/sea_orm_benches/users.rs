//! SeaORM Entity. Generated by sea-orm-codegen 0.2.4
use sea_orm::entity::prelude::*;
use std::convert::TryInto;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "users")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
    pub hair_color: Option<String>,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::posts::Entity")]
    Posts,
}

impl Related<super::posts::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::Posts.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
