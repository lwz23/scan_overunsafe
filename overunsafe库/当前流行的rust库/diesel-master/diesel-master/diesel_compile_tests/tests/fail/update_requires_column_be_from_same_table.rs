extern crate diesel;

use diesel::*;

table! {
    users {
        id -> Integer,
        name -> VarChar,
    }
}

table! {
    posts {
        id -> Integer,
        title -> VarChar,
    }
}

allow_tables_to_appear_in_same_query!(users, posts);

fn main() {
    use self::users::dsl::*;

    let command = update(users).set(posts::title.eq("Hello"));
    let command = update(users).set(name.eq(posts::title));
}
