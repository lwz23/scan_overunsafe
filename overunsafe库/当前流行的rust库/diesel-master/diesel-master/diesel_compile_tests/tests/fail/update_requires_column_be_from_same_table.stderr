error[E0271]: type mismatch resolving `<Grouped<Eq<title, Bound<Text, &str>>> as AsChangeset>::Target == table`
  --> tests/fail/update_requires_column_be_from_same_table.rs:24:37
   |
24 |     let command = update(users).set(posts::title.eq("Hello"));
   |                                 --- ^^^^^^^^^^^^^^^^^^^^^^^^ expected `users::table`, found `posts::table`
   |                                 |
   |                                 required by a bound introduced by this call
   |
   = note: `posts::table` and `users::table` have similar names, but are actually distinct types
note: `posts::table` is defined in module `crate::posts` of the current crate
  --> tests/fail/update_requires_column_be_from_same_table.rs:12:1
   |
12 | / table! {
13 | |     posts {
14 | |         id -> Integer,
15 | |         title -> VarChar,
16 | |     }
17 | | }
   | |_^
note: `users::table` is defined in module `crate::users` of the current crate
  --> tests/fail/update_requires_column_be_from_same_table.rs:5:1
   |
5  | / table! {
6  | |     users {
7  | |         id -> Integer,
8  | |         name -> VarChar,
9  | |     }
10 | | }
   | |_^
note: the method call chain might not have had the expected associated types
  --> tests/fail/update_requires_column_be_from_same_table.rs:24:50
   |
24 |     let command = update(users).set(posts::title.eq("Hello"));
   |                                     ------------ ^^^^^^^^^^^ `AsChangeset::Target` is `table` here
   |                                     |
   |                                     this expression has type `title`
note: required by a bound in `UpdateStatement::<T, U>::set`
  --> $DIESEL/src/query_builder/update_statement/mod.rs
   |
   |     pub fn set<V>(self, values: V) -> UpdateStatement<T, U, V::Changeset>
   |            --- required by a bound in this associated function
...
   |         V: changeset::AsChangeset<Target = T>,
   |                                   ^^^^^^^^^^ required by this bound in `UpdateStatement::<T, U>::set`
   = note: this error originates in the macro `table` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0271]: type mismatch resolving `<table as AppearsInFromClause<table>>::Count == Once`
  --> tests/fail/update_requires_column_be_from_same_table.rs:25:33
   |
25 |     let command = update(users).set(name.eq(posts::title));
   |                                 ^^^ expected `Never`, found `Once`
   |
note: required for `posts::columns::title` to implement `AppearsOnTable<users::table>`
  --> tests/fail/update_requires_column_be_from_same_table.rs:15:9
   |
15 |         title -> VarChar,
   |         ^^^^^
   = note: associated types for the current `impl` cannot be restricted in `where` clauses
   = note: required for `diesel::expression::operators::Eq<users::columns::name, posts::columns::title>` to implement `AsChangeset`
